lib/Backend/Lower.cpp

```diff
@@ -29161,3 +29111,191 @@ Lowerer::LowerCheckUpperIntBound(IR::Instr * instr)
     return instrPrev;
 }
 #endif
+
+Lowerer::LowerGeneratorHelper::LowerGeneratorHelper(Func* func, Lowerer* lowerer, LowererMD& lowererMD):
+    func(func), lowerer(lowerer), lowererMD(lowererMD) {}
+
+void Lowerer::LowerGeneratorHelper::InsertNullOutGeneratorFrameInEpilogue(IR::LabelInstr* epilogueLabel)
+{
+    IR::Instr* insertionPoint = epilogueLabel->m_next;
+
+    // null out frame pointer on generator object to signal completion to JavascriptGenerator::CallGenerator
+    // s = MOV prm1
+    // s[offset of JavascriptGenerator::frame] = MOV nullptr
+    StackSym* symSrc = StackSym::NewImplicitParamSym(3, this->func);
+    this->func->SetArgOffset(symSrc, LowererMD::GetFormalParamOffset() * MachPtr);
+    IR::SymOpnd* srcOpnd = IR::SymOpnd::New(symSrc, TyMachPtr, this->func);
+    IR::RegOpnd* dstOpnd = IR::RegOpnd::New(TyMachReg, this->func);
+    dstOpnd->SetReg(RegRSI); // callee-save register
+    InsertMove(dstOpnd, srcOpnd, insertionPoint);
+
+    IR::IndirOpnd* indirOpnd = IR::IndirOpnd::New(dstOpnd, Js::JavascriptGenerator::GetFrameOffset(), TyMachPtr, this->func);
+    IR::AddrOpnd* addrOpnd = IR::AddrOpnd::NewNull(this->func);
+    InsertMove(indirOpnd, addrOpnd, insertionPoint);
+}
+
+void
+Lowerer::LowerGeneratorHelper::InsertBailOutForElidedYield()
+{
+    IR::LabelInstr* bailOutNoSaveLabel = nullptr;
+
+    this->func->MapUntilYieldOffsetResumeLabels([this, &bailOutNoSaveLabel](int, const YieldOffsetResumeLabel& yorl)
+    {
+        if (yorl.Second() == nullptr)
+        {
+            if (bailOutNoSaveLabel == nullptr)
+            {
+                bailOutNoSaveLabel = IR::LabelInstr::New(Js::OpCode::Label, this->func);
+            }
+
+            return true;
+        }
+
+        return false;
+    });
+
+    // Insert the bailoutnosave label somewhere along with a call to BailOutNoSave helper
+    if (bailOutNoSaveLabel != nullptr)
+    {
+        IR::Instr* exitPrevInstr = this->GetEpilogueForReturnStatements()->m_prev;
+        IR::LabelInstr* exitTargetInstr = this->GetEpilogueForBailOut();
+
+        bailOutNoSaveLabel->m_hasNonBranchRef = true;
+        bailOutNoSaveLabel->isOpHelper = true;
+
+        IR::Instr* bailOutCall = IR::Instr::New(Js::OpCode::Call, this->func);
+
+        exitPrevInstr->InsertAfter(bailOutCall);
```

Comment:
nit: using InsertBefore is usually easier to reason about, since insertAfter you kind of need to insert the instrs backwards. so instead you could do exitInstr->InsertBefore(bailOutNoSaveLabel)/bailOutCall/branch